src/internal/observable/fromObservable.ts

```diff
@@ -0,0 +1,28 @@
+import { Observable, Subscribable } from '../Observable';
+import { IScheduler } from '../Scheduler';
+import { Subscription } from '../Subscription';
+import { observable as Symbol_observable } from '../symbol/observable';
+
+// tslint:disable-next-line:no-any TS is unable to type [Symbol.observable]
+export function fromObservable<T>(input: any, scheduler: IScheduler) {
+  if (!scheduler) {
+    return new Observable<T>(subscriber => {
+      const observable: Subscribable<T> = input[Symbol_observable]();
+      const sub = observable.subscribe(subscriber);
+      return () => sub && sub.unsubscribe();
+    });
+  } else {
+    return new Observable<T>(subscriber => {
+      const sub = new Subscription();
+      sub.add(scheduler.schedule(() => {
```

Comment:
do we need to create new subscription instead of returning subscription as-is from `Scheduler::schedule()`?

Reply:
yes, because I'm closing over it to manage child subscriptions below. (line 21, for example)